--- src/formats.c.orig
+++ src/formats.c
@@ -69,6 +69,8 @@
   CHECK(wav   , 0, 4, "RIFX" , 8,  4, "WAVE")
   CHECK(wav   , 0, 4, "RF64" , 8,  4, "WAVE")
   CHECK(aiff  , 0, 4, "FORM" , 8,  4, "AIFF")
+  CHECK(dsdiff, 0, 4, "FRM8" , 8,  4, "DSD ")
+  CHECK(dsf   , 0, 4, "DSD " , 0,  0, "")
   CHECK(aifc  , 0, 4, "FORM" , 8,  4, "AIFC")
   CHECK(8svx  , 0, 4, "FORM" , 8,  4, "8SVX")
   CHECK(maud  , 0, 4, "FORM" , 8,  4, "MAUD")
